bot/exts/code_jams/_creation_utils.py

```diff
@@ -127,3 +128,19 @@ async def _add_team_leader_roles(members: list[dict[str: discord.Member, str: bo
     for entry in members:
         if entry["is_leader"]:
             await entry["member"].add_roles(team_leaders)
+
+
+async def pin_message(message: discord.Message, ctx: commands.Context, unpin: bool = False) -> None:
+    """A discord.py helper to pin or unpin messages and handle possible exceptions."""
+    try:
+        if unpin:
+            await message.unpin(reason="Code Jam organization")
+            await ctx.reply(":white_check_mark: The message has been unpinned!")
+        else:
+            await message.pin(reason="Code Jam organization")
+            await ctx.reply(":white_check_mark: The message has been pinned!")
+    except discord.HTTPException as err:
+        await ctx.reply(
+            f"Something went wrong {', you might have reached the 50 pins per channel limit!' if not unpin else ''}"
+        )
+        log.trace(f"Something went wrong when pinning a CJ message: {err}")
```

Comment:
This pin wrapper can be re-used from the bot, as it comes with more error handling.

https://github.com/python-discord/bot/blob/main/bot/exts/help_channels/_message.py#L264-L286